src/Tribe/Views/V2/View.php

```diff
@@ -2194,23 +2194,36 @@ protected function should_show_recent_past_events_view() {
 	 */
 	protected function setup_additional_views( array $events = [], array $template_vars = [] ) {
 
+		// Only return if on the current or upcoming view and not in the past.
+		if ( 'past' === $this->context->get( 'event_display_mode' ) ) {
+			return;
+		}
+
 		$manager = tribe( Manager::class );
 		$default_slug = $manager->get_default_view_option();
 
-		// Only recent Past Events to only the default view.
+		// Show Recent Past Events only on the default view.
 		if ( $this->get_slug() !== $default_slug ) {
 			return;
 		}
 
+		$latest = tribe_events_latest_date();
+		$now = $this->context->get( 'now' );
```

Comment:
:warning: Equals sign not aligned with surrounding assignments; expected 4 spaces but found 1 space

<sub>hash: ca874947acbfbe8bd48db0f6ff8a7b84</sub>